File: fitness_app/lib/models/message.dart
class Message {
  final String id;
  final String conversationId;
  final String sender;
  final String content;
  final DateTime createdAt;

  Message({required this.id, required this.conversationId, required this.sender, required this.content, required this.createdAt});

  factory Message.fromJson(Map<String, dynamic> json) => Message(
    id: json['id'],
    conversationId: json['conversationId'],
    sender: json['sender'],
    content: json['content'],
    createdAt: DateTime.parse(json['createdAt']),
  );

  Map<String, dynamic> toJson() {
    return {
      "id": id,
      "conversationId": conversationId,
      "sender": sender,
      "content": content,
      "createdAt": createdAt
    };
  }
}

File: src/dotnet/models/Message.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace dotnet.Models
{
    public class Message
    {
        [Key]
        public Guid Id { get; set; }
        [Required]
        public Guid ConversationId { get; set; }
        [ForeignKey("ConversationId")]
        public Conversation Conversation { get; set; } = null!;
        [Required]
        public string Sender { get; set; } = "user"; // or "assistant"
        [Required]
        public string Content { get; set; } = "";
        [Required]
        public DateTime CreatedAt { get; set; }
    }
}